### PR TITLE
add Spark-TTS

### PR DESCRIPTION
**Pull request by LiTingyu1997**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/sparktts/example/results/20250225113521.wav
Comment (SamitHuang) [inline examples/sparktts/example/results/20250225113521.wav:1]: rm this file

File: examples/sparktts/src/demos/jay/jay.wav
Comment (SamitHuang) [inline examples/sparktts/src/demos/jay/jay.wav:1]: just keep one or two wav files for the demo running? to reduce repo size

File: examples/sparktts/README.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+<div align="center">
+    <h1>
+    Spark-TTS
+    </h1>
+    <p>
+    Official PyTorch code for inference of <br>
```
Comment (SamitHuang) [inline examples/sparktts/README.md:6]: rm `Official PyTorch code for` and check the other parts

File: examples/sparktts/README.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+  <tr>
+    <td align="center"><b>Inference Overview of Voice Cloning</b><br><img src="src/figures/infer_voice_cloning.png" width="80%" /></td>
+  </tr>
+  <tr>
+    <td align="center"><b>Inference Overview of Controlled Generation</b><br><img src="src/figures/infer_control.png" width="80%" /></td>
+  </tr>
+</table>
+
+
+## 🚀 News
```
Comment (SamitHuang) [inline examples/sparktts/README.md:38]: remove News. Or just mention "mindspore implementation of SpartTTS is released"

File: mindone/utils/weight_norm.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def norm_except_dim(v, pow, dim):
         output_size = (1,) * (v.ndim - 1) + (v.shape[v.ndim - 1])
         return ops.norm(v.view((-1, v.shape[v.ndim - 1])), pow, 0).view(output_size)
     else:
-        return norm_except_dim(v.swapaxes(0, dim), pow, dim).swapaxes(0, dim)
+        return norm_except_dim(v.transpose(0, v.ndim - 1), pow, 0).transpose(0, v.ndim - 1)
```
Comment (SamitHuang) [inline mindone/utils/weight_norm.py:15]: why change to transpose?

File: mindone/transformers/modeling_utils.py [hunk 4]
```diff
@@ -80,6 +80,15 @@ def _get_pt2ms_mappings(m):
             mappings[f"{name}.weight"] = f"{name}.weight", lambda x: ms.Parameter(
                 ops.expand_dims(x, axis=-2), name=x.name
             )
+            if "weight_norm_cell" in name:
```
Comment (SamitHuang) [inline mindone/transformers/modeling_utils.py:83]: it seems not a common name mapping for all models.
is it caused by the variable naming difference when writing qwen2.5lm? if so, it's better to just change the layer name in qwen2.5

File: examples/sparktts/README.md (hunk 1 above)
Comment (LiTingyu1997) [reply examples/sparktts/README.md]: done

File: mindone/transformers/modeling_utils.py (hunk 4 above)
Comment (LiTingyu1997) [reply mindone/transformers/modeling_utils.py:83]: This check is necessary if the model uses weight_norm, as it is a general requirement and should be kept

File: mindone/utils/weight_norm.py (hunk 3 above)
Comment (LiTingyu1997) [reply mindone/utils/weight_norm.py:15]: 1.Modified the calculation of weight_norm when dim > 2. Wav2Vec2 triggered this scenario, where numerical inaccuracies were detected, leading to a logic adjustment to ensure accurate computation.

2.Aligned with PyTorch's use of transpose.

File: examples/sparktts/README.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+
+## **Demos**
+
+Here are some demos generated by Spark-TTS using zero-shot voice cloning.
+
+``` sh
+python -m cli.inference \
+    --text "我觉得这种运动其实不是说靠机会的，我觉得对每个人来讲，像我们歌手来讲，我觉得其实都是你要自己去努力，然后才可以达到自己的梦想。" \
+    --device 0 \
```
Comment (HaoyangLee) [inline examples/sparktts/README.md:109]: There is no `--device` in our arguments, delete it in all the command lines listed in the README.

File: examples/sparktts/README.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+git clone https://huggingface.co/SparkAudio/Spark-TTS-0.5B pretrained_models/Spark-TTS-0.5B
+```
+
+**Basic Usage**
+
+You can simply run the demo with the following commands:
+``` sh
+cd example
+bash infer.sh
```
Comment (HaoyangLee) [inline examples/sparktts/README.md:86]:
```suggestion
bash example/infer.sh
```

**Review by HaoyangLee — CHANGES_REQUESTED** [review]

File: examples/sparktts/README.md (hunk 5 above)
Comment (LiTingyu1997) [reply examples/sparktts/README.md]: done

**Review by vigo999 — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]
